src/v2/providers/schedule.ts

```diff
@@ -0,0 +1,198 @@
+  };
+  const opts = { ...args };
+  delete (opts as any).schedule;
+  delete (opts as any).timeZone;
+  delete (opts as any).retryCount;
+  delete (opts as any).maxRetryDuration;
+  delete (opts as any).minBackoffDuration;
+  delete (opts as any).maxBackoffDuration;
+  delete (opts as any).maxDoublings;
+
+  scheduleArgs.opts = opts;
+  return scheduleArgs as ScheduleArgs;
+}
+
+/** The Cloud Function type for Schedule triggers. */
+export interface ScheduleFunction extends HttpsFunction {
+  __requiredAPIs?: ManifestRequiredAPI[];
+  run(data: express.Request): any | Promise<any>;
+}
+
+/** Options that can be set on a Schedule trigger. */
+export interface ScheduleOptions extends options.GlobalOptions {
+  /** The schedule, in Unix Crontab or AppEngine syntax. */
+  schedule: string;
+
+  /** The timezone that the schedule executes in. */
+  timeZone?: timezone;
+
+  /** The number of retry attempts for a failed run. */
+  retryCount?: number;
+
+  /** The time limit for retrying. */
+  maxRetryDuration?: string;
+
+  /** The minimum time to wait before retying. */
+  minBackoffDuration?: string;
+
+  /** The maximum time to wait before retrying. */
+  maxBackoffDuration?: string;
+
+  /** The time between will double max doublings times. */
+  maxDoublings?: number;
+}
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param schedule - The schedule, in Unix Crontab or AppEngine syntax.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  schedule: string,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction;
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param options - Options to set on scheduled functions.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  options: ScheduleOptions,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction;
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param args - Either a schedule or an object containing function options.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  args: string | ScheduleOptions,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction {
+  const separatedOpts = getOpts(args);
+
+  const func: any = async (
+    req: express.Request,
+    res: express.Response
+  ): Promise<any> => {
+    try {
+      await handler(req);
+    } catch (err) {
+      logger.warn((err as Error).message);
+    }
+
+    res.setHeader('Content-Type', 'application/json');
+    res.status(200).send();
```

Comment:
why do we always respond w/ 200? Doesn't this mean that we won't retry execution w/ errors?

Reply:
The current behavior for v1 is that a function error will not retry the execution. A retry will occur when the scheduler job fails. I opted for consistency and just return a 200. Let me know if you think we should change this up for v2.

Reply:
Hm following footsteps of v1 scheduled function seems reasonable, though I would guess having scheduled functions automatically retry on transient errors to be pretty useful. This is something we explicitly set for event functions via `failurePolicy` and I can see it being applied here too.

But if you think that is a feature that doesn't need to be implemented v2 scheduled function, that sounds fine with me. Maybe we'll get a FR for it one day, and we can implement it then?

Reply:
SG, updated to send a 500 if we catch an error.